src/components/field/field-name-content.tsx

```diff
@@ -0,0 +1,77 @@
+import { styled } from 'storybook/internal/theming';
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+import { ellipsisTruncation } from '@/styles/styles';
+import { DEFAULT_FIELD_HEIGHT } from '@/utilities/constants';
+
+const InnerFieldName = styled.div`
+  width: 100%;
+  min-height: ${DEFAULT_FIELD_HEIGHT}px;
+  ${ellipsisTruncation}
+`;
+
+const InlineInput = styled.input`
+  border: none;
+  background: none;
+  height: ${DEFAULT_FIELD_HEIGHT}px;
+  color: inherit;
+  font-size: inherit;
+  font-family: inherit;
+  font-style: inherit;
+`;
+
+interface FieldNameProps {
+  name: string;
+  isEditable?: boolean;
+  onChange?: (newName: string) => void;
+  onBlur?: () => void;
+}
+
+export const FieldNameContent = ({ name, isEditable, onChange }: FieldNameProps) => {
```

Comment:
Would it be better to have two seperate components, one which is editable and the other (the current flow) which is not? At the moment we're passing in quite a few (optional) props which require a conditional check, especially in that `useEffect`